mmaction/core/bbox/samplers/base_sampler.py

```diff
@@ -0,0 +1,87 @@
+from abc import ABCMeta, abstractmethod
+
+import torch
+
+from .sampling_result import SamplingResult
+
+
+class BaseSampler(metaclass=ABCMeta):
+    """Base class for bbox_samplers.
+
+    All datasets to sample bboxes should subclass it.
+    All subclasses should overwrite:
+
+    - Methods:`_sample_pos`, sample_positive_bboxes
+    - Methods:`_sample_neg`, sample_negative_bboxes
+
+    Args:
+        num (int): Number of bboxes to sample.
+        pos_fraction (float): Fraction of positive bboxes (expected).
+        neg_pos_ub (float): The upper bound of #negative / #positive. If set as
+            negative, the constraint will be ignored. Default: -1.
+        add_gt_as_proposals (bool): Also add gt bboxes as proposals.
+            Default: True.
+    """
+
+    def __init__(self,
+                 num,
+                 pos_fraction,
+                 neg_pos_ub=-1,
+                 add_gt_as_proposals=True):
+
+        self.num = num
+        self.pos_fraction = pos_fraction
+        self.neg_pos_ub = neg_pos_ub
+        self.add_gt_as_proposals = add_gt_as_proposals
+
+    @abstractmethod
+    def _sample_pos(self, assign_result, num_expected):
+        pass
```

Comment:
`pass` is redundant

Reply:
For all abstract methods in this repo, `pass` is added. I think the body is needed even for an abstract function, the interpreter may throw out an error if we remove `pass` here.